torch/csrc/distributed/autograd/context/container.h

```diff
@@ -82,6 +82,12 @@ class TORCH_API DistAutogradContainer {
   // Returns the number of autograd contexts in the container.
   size_t numAutogradContexts() const;
 
+  // Retrieves the autograd context, if it exists, and sets the autograd error
+  // on that context when there is an autograd failure.
+  void setAutogradContextErrorIfPresent(
```

Comment:
Doesn't seem right to add this method to the container. Can we do this in request_callback_impl.cpp instead?

1. Retrieve the context from the container.
2. If it exists, retrieve the graph task. If the graph task is valid set an exception on it.
3. If the context doesn't exist or graph task is not valid, just propagate the error to all other nodes.